### ISSUE TITLE
Wrong r12 offset in FreeRTOS stack frame

### ISSUE THREAD

**Issue by newbrain** [pyocd/pyOCD#1043]
Offset of r12 in FreeRTOS stack frame dictionary at line 102 in pyocd/rtos/freertos.py is not correct:
https://github.com/pyocd/pyOCD/blob/f146817ad2233d247e70fbba6adfe03b2c3fd8cc/pyocd/rtos/freertos.py#L101-L103

The correct value should be 52, (42 is, oddly, not the correct answer here).

**Comment by flit** [comment]
Oops, thanks for the report. 

> (42 is, oddly, not the correct answer here)

🤣 


Fyi, I have an update to the FreeRTOS plugin that supports FreeRTOS 10.x. It needs more testing, and I haven't yet checked whether it breaks compatibility with FreeRTOS 8.x/9.x. The branch is https://github.com/flit/pyOCD/tree/feature/freertos_10.

**Comment by newbrain** [comment]
> Fyi, I have an update to the FreeRTOS plugin that supports FreeRTOS 10.x.
Interesting, thanks.

I'm working on a ThreadX (AzureRTOS) plugin, that's why I've noticed the typo.
It seems to work but I need to set up a compliant development environment and to test it according to the Contributing.md document, till now I've just added the new code on my installation(s).
If all goes well, I'll make a pull request in 1-2 weeks.
